lib/src/data/course.dart

```diff
@@ -0,0 +1,29 @@
+import "utils.dart";
+import "chat.dart";
+
+extension type CourseId(String value) {}
+
+class Course {
+  final String department;
+  final int course_number;
+  final String subject;
+  Chat chat;
```

Comment:
```suggestion
  final ChatID chat;
```

Reply:
We should still be using `ChatID`s instead of just plain `Chat`, since that's a lot of information to embed directly into the course, and it'll make things more complicated later if we want to subscribe to changes in the chats (and not, eg, deal with all the course data too)